### PR TITLE
Subscriber configurability, sensitive and validated settings, documentation

### LINKED ISSUE #261: szChannelID and szChannelName1 should be user-settable properties
## Bug report
szChannelID and szChannelName1 should be user-settable properties, but they're hard coded string constants, so you'd have to recompile it to target it to a different channel.  They should be handled like zip code.  We probably need a way to set the API key for weather, too...

**Problem**
I can't make a Mr. Beast counter without recompiling.

### PR DESCRIPTION
**Pull request by rbergen**
## Description

This:

- Makes the YouTube channel GUID and name, as used by the Subscriber effect, configurable using DeviceConfig
- Adds the ability to have individual settings validated by POSTing them to the /settings/validated sub-endpoint
- Hides sensitive settings when returning the configuration on the /settings endpoint
- Adds the `override` specifier to JSON-related functions
- Documents the API published by the built-in webserver
- Provides a Postman collection file to facilitate the use of the on-device API

Because of the first point, this fixes #261.

## Contributing requirements

* [X] I read the contribution guidelines in [CONTRIBUTING.md](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/CONTRIBUTING.md).
* [X] I understand the BlinkenPerBit metric, and maximized it in this PR.
* [X] I selected `main` as the target branch.
* [X] All code herein is subjected to the license terms in [COPYING.txt](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/COPYING.txt).

### CODE REVIEW

File: REST_API.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+# On-board REST-like API <!-- omit in toc -->
```
Comment (davepl) [inline REST_API.md:1]: Appreciate you documenting this!

**Review by davepl — APPROVED** [review]
LGTM!

File: include/deviceconfig.h [hunk 2]
```diff
@@ -110,6 +116,8 @@ class DeviceConfig : public IJSONSerializable
         SetIfPresentIn(jsonObject, openWeatherApiKey, OpenWeatherApiKeyTag);
         SetIfPresentIn(jsonObject, use24HourClock, Use24HourClockTag);
         SetIfPresentIn(jsonObject, useCelsius, UseCelsiusTag);
+        SetIfPresentIn(jsonObject, youtubeChannelGuid, YouTubeChannelGuidTag);
+        SetIfPresentIn(jsonObject, youtubeChannelName1, YouTubeChannelName1Tag);
```
[the remark this answers is not in the document]
Comment (rbergen) [reply include/deviceconfig.h:130]: I adopted this from the old code. I had the same question, but as it was used a few times I decided to retain the name with 1 added.

File: REST_API.md (hunk 1 above)
Comment (rbergen) [reply REST_API.md:1]: You're welcome!

**Pull request merged**